cmd/crdvalidator/README.md

```diff
@@ -0,0 +1,116 @@
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+    core.rukpak.io/safe-upgrade-validation: disabled # <--
+# 
+# ...
+#
+```
+
+
+### What is protected
+There are three main cases `crdvalidator` is checking against with every specified `CRD` upgrade. These are where a new `CRD`:
+1. Removes a version that the old `CRD` had and it must ensured that nothing is stored at the removed version.
+2. Changes a version that old `CRD` had and it must validate existing CRs against the new schema.
+3. Adds a version that old `CRD` does not have. In this case `crdvalidator` checks if the conversion strategy is:
+    - None, then ensures that existing CRs validate with new schema.
+    - Webhook, then allow update (assuming webhook handles conversion correctly)
+
+#### Update the schema that invalidates existing CRs
+Say that you have installed a `CRD` onto the cluster with a Kind of `Sample` that has a single property of `examplearray`. 
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+    core.rukpak.io/safe-upgrade-validation: enabled
+  creationTimestamp: null
+  name: templates.combo.io
+#
+# ...
+#
+spec:
+    versions:
+    - name: v1alpha1
+        schema:
+        openAPIV3Schema:
+            properties:
+                examplearray:
+                    description: Examplearray is an array of strings to demonstrate crd validation logic.
+                    type: array
+                    minItems: 1
+                    items:
+                    type: string
+```
+
+With this installed onto the cluster, you go ahead and create a `CR` of a `Sample` onto the cluster. If you were to then update that `CRD`'s schema in such a way that that `CR` you created was invalidated, like so:
+
+```yaml
+#
+# ...
+#
+spec:
+    versions:
+    - name: v1alpha1
+        schema:
+        openAPIV3Schema:
+            properties:
+                examplearray:
+                    description: Examplearray is an array of strings to demonstrate crd validation logic.
+                    type: array
+                    minItems: 100
+                    items:
+                    type: string
+```
+
+
+`crdvalidator` will prevent the update from occurring if the existing `CR` does not have at least 100 items in the `examplearray` field.
+
+## Running locally
+
+`crdvalidator` is included with every default installation of RukPak. To run this locally, simply run the make target with a kind cluster started.
+
+```console
+$ make run
+```
+
+You can additionally run the `crdvalidator`'s e2e suite along side of RukPak's.
+
+```console
+$ make e2e
+```
+
+If you would like to install the `crdvalidator` onto your current cluster without RukPak, you can easily do this with the appropriate make target.
```

Comment:
Should we mention the cert-manager dependency? Right now we refactored the Makefile to pull the cert-manager dependency out of the install path of the individual webhook. Might be helpful to mention `make cert-manager` to install cert-manager as a prereq.